cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/PullFilter.java

```diff
@@ -148,4 +146,10 @@ public String toQueryString() {
             return Misc.join("&", queries);
         }
     }
+
+    protected void setParameters(Map<String, String> parameters) {
+        Map<String, String> internalParams = new HashMap<String, String>();
```

Comment:
Since `parameters` isn't exposed with a getter, we don't need to make it unmodifiable any more. This can be a straightforward setter method.

Reply:
I'm not sure of this. If the application keeps a reference to the parameter map those parameters can be modified without our control.

Reply:
There is a getter over parameters in PullFilter. I think we require the unmodifiable list